### PR TITLE
Flutter RUM Base 

### PR DESCRIPTION
**Pull request by NasarTarique**
The following Instrumentation for flutter mobile applications are included in these Initial Commits.

- Mobile App Performance
  - CPU usage
  - memory usage
  - cold/warm start
  - ANR ( android)
- Flutter Errors & Exceptions
- Events
  - session start
  - route changes
  - user interaction
- HTTP network info
  - Load Duration , method , resource/type , request/response size
- Rum Asset Bundle
  - Asset Size, Load time
- Custom events, logs, measurement, error
- Offline Caching of Events

### CODE REVIEW

**Review by robert-northmind — COMMENTED** [review]
Wow, this is really great! Thanks again for this contribution.
I left some minor comments/thoughts.

**Review by robert-northmind — APPROVED** [review]

File: packages/offline_transport/lib/offline_transport.dart [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+
+      if (_maxCacheDuration != null &&
+          currentTime - timestamp > _maxCacheDuration!.inMilliseconds) {
+        continue;
+      } else {
+        await sendCachedData(payload);
+      }
+    }
+
+    if (remainingLines.isEmpty) {
```
[the remark this answers is not in the document]
Comment (NasarTarique) [reply packages/offline_transport/lib/offline_transport.dart]: I don't think it's used anywhere so i've removed it .

File: packages/offline_transport/pubspec.yaml
[the remark this answers is not in the document]
Comment (NasarTarique) [reply packages/offline_transport/pubspec.yaml:1]: yes it was just to keep dependencies separate for users who don't want to use it

File: packages/rum_sdk/pubspec.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
+publish_to: none
+name: rum_sdk
```
[the remark this answers is not in the document]
Comment (NasarTarique) [reply packages/rum_sdk/pubspec.yaml:2]: yes, i think that would make more sense

File: packages/rum_sdk/lib/rum_sdk.dart [hunk 3]
```diff
@@ -0,0 +1,21 @@
+library;
+
+
+export 'rum_flutter.dart'
+    if (dart.library.html) 'package:rum_sdk/rum_web.dart';
```
[the remark this answers is not in the document]
Comment (NasarTarique) [reply packages/rum_sdk/lib/rum_sdk.dart]: That was for having a separate initializer for web but we can remove it for now

**Review by robert-northmind — APPROVED** [review]

**Pull request merged**